A notebook to perform sequence clustering with this data
Source: https://www.ncbi.nlm.nih.gov/sra/SRX26494648%5Baccn%5D


In [4]:
import subprocess

## FastQ / preproccesing etc

In [9]:
# will need to alter this when switching to codespace 

# local path
! /Users/milton/anaconda3/bin/fastqc /Users/milton/Documents/University/Dissertation/git/DNA-Data/SRR31112377.fastq -o /Users/milton/Documents/University/Dissertation/git/DNA-Data/fastq_reports/

# codespace path
# ! /usr/bin/fastqc /workspaces/DNA-Data/SRR31112377.fastq -o /workspaces/DNA-Data/fastq_reports/

null
Started analysis of SRR31112377.fastq
Approx 5% complete for SRR31112377.fastq
Approx 10% complete for SRR31112377.fastq
Approx 15% complete for SRR31112377.fastq
Approx 20% complete for SRR31112377.fastq
Approx 25% complete for SRR31112377.fastq
Approx 30% complete for SRR31112377.fastq
Approx 35% complete for SRR31112377.fastq
Approx 40% complete for SRR31112377.fastq
Approx 45% complete for SRR31112377.fastq
Approx 50% complete for SRR31112377.fastq
Approx 55% complete for SRR31112377.fastq
Approx 60% complete for SRR31112377.fastq
Approx 65% complete for SRR31112377.fastq
Approx 70% complete for SRR31112377.fastq
Approx 75% complete for SRR31112377.fastq
Approx 80% complete for SRR31112377.fastq
Approx 85% complete for SRR31112377.fastq
Approx 90% complete for SRR31112377.fastq
Approx 95% complete for SRR31112377.fastq
Analysis complete for SRR31112377.fastq


### Results from report (viewable via html file)
- Per Base Sequence Content (Yellow Warning)
    - Indication that nucleotide is not balanced across the sequence
    - A,T,G,C freqs should be pretty much equal
    - Could be bias in library prep or contamination
- Per Sequence GC Content (Red Cross)
    - GC distro does not match GC content
    - skewed distro could be contamination from another organism or technical bias
- Sequence Length Distribution (Yellow Warning)
    - Reads have variable lengths
    - ! shouldn't be pre-trimmed double check that !
- Sequence Duplication Levels (Red Cross)
    - High duplication suuggests artifacts or low diversity.
    - Bit of an issue (perhaps remove duplicates?)
    
Everything else seems ok.

In [16]:
# NOTE: this code cannot be run without fastp installation, install in codespace!
!fastp -i SRR31112377.fastq -o fastp_reports/output.fastq -h fastp_reports/report.html -j fastp_reports/report.json

Detecting adapter sequence for read1...
No adapter detected for read1

Read1 before filtering:
total reads: 1389162
total bases: 151004417
Q20 bases: 151004417(100%)
Q30 bases: 151004417(100%)

Read1 after filtering:
total reads: 1388473
total bases: 150980302
Q20 bases: 150980302(100%)
Q30 bases: 150980302(100%)

Filtering result:
reads passed filter: 1388473
reads failed due to low quality: 0
reads failed due to too many N: 689
reads failed due to too short: 0
reads with adapter trimmed: 0
bases trimmed due to adapters: 0

Duplication rate (may be overestimated since this is SE data): 16.9656%

JSON report: fastp_reports/report.json
HTML report: fastp_reports/report.html

fastp -i SRR31112377.fastq -o fastp_reports/output.fastq -h fastp_reports/report.html -j fastp_reports/report.json 
fastp v0.24.0, time used: 7 seconds


### Fastp results
1. No adapter detected = no contamination = good
2. Before n after loss of 689 reads due to too many Ns 
3. 100% Q20 & Q30 bases so pretty good quality data
4. Duplication rate 16.97% this might be expected? not sure 